src/main/java/org/sonar/plugins/github/GitHubPluginConfiguration.java

```diff
@@ -117,4 +133,60 @@ public boolean tryReportIssuesInline() {
     return !settings.getBoolean(GitHubPlugin.GITHUB_DISABLE_INLINE_COMMENTS);
   }
 
+    /**
+     * Checks if a proxy was passed with command line parameters or configured in the system.
+     * If only an HTTP proxy was configured then it's properties are copied to the HTTPS proxy (like SonarQube configuration)
+     * @return True iff a proxy was configured to be used in the plugin.
+     */
+  public boolean isProxyConnectionEnabled() {
+    if (System.getProperty(HTTP_PROXY_HOSTNAME) != null || System.getProperty(HTTPS_PROXY_HOSTNAME) != null ||
+            System.getProperty(PROXY_SOCKS_HOSTNAME) != null)
+    {
+        return true;
+    }
+    return false;
+  }
+
+
+  public Proxy getHttpProxy() {
+    try
+    {
+        if(System.getProperty(HTTP_PROXY_HOSTNAME) != null && System.getProperty(HTTPS_PROXY_HOSTNAME) == null)
+        {
+            System.setProperty(HTTPS_PROXY_HOSTNAME, System.getProperty(HTTP_PROXY_HOSTNAME));
+            System.setProperty(HTTPS_PROXY_PORT, System.getProperty(HTTP_PROXY_PORT));
+
+        }
+
+        String proxyUser = System.getProperty(HTTP_PROXY_USER);
+        String proxyPass = System.getProperty(HTTP_PROXY_PASS);
+
+        if(proxyUser != null && proxyPass != null)
+        {
+            Authenticator.setDefault(
+                    new Authenticator() {
+                        @Override
+                        public PasswordAuthentication getPasswordAuthentication() {
+                            return new PasswordAuthentication(
+                                    proxyUser, proxyPass.toCharArray());
+                        }
+                    }
+            );
+        }
+
+        Proxy selectedProxy = ProxySelector.getDefault().select(new URI(endpoint())).get(0);
+
+        if(selectedProxy.type() == Proxy.Type.DIRECT)
+        {
+            LOG.debug("There was no suitable proxy found to connect to GitHub - direct connection is used ");
+        }
+
+        LOG.info("A proxy has been configured - " + selectedProxy.toString());
+        return selectedProxy;
+    } catch (URISyntaxException e)
+        {
+            throw new IllegalArgumentException("Unable to perform GitHub WS operation - endpoint in wrong format: " + endpoint() + e.getMessage());
```

Comment:
Oups, missed that one. Don't append e.getMessage(). Instead you must wrap the original exception (passing e as second argument).